package sopimusrekisteri.test;
// Generated by ComTest BEGIN
import static org.junit.Assert.*;
import org.junit.*;
import sopimusrekisteri.*;
// Generated by ComTest END

/**
 * Test class made by ComTest
 * @version 2020.02.19 23:35:03 // Generated by ComTest
 *
 */
@SuppressWarnings({ "all" })
public class PelaajatTest {



  // Generated by ComTest BEGIN
  /** 
   * testLisaa35 
   * @throws SailoException when error
   */
  @Test
  public void testLisaa35() throws SailoException {    // Pelaajat: 35
    Pelaajat pelaajat = new Pelaajat(); 
    Pelaaja p1 = new Pelaaja(), p2 = new Pelaaja(); 
    assertEquals("From: Pelaajat line: 39", 0, pelaajat.getLkm()); 
    pelaajat.lisaa(p1); assertEquals("From: Pelaajat line: 40", 1, pelaajat.getLkm()); 
    pelaajat.lisaa(p2); assertEquals("From: Pelaajat line: 41", 2, pelaajat.getLkm()); 
    pelaajat.lisaa(p2); assertEquals("From: Pelaajat line: 42", 3, pelaajat.getLkm()); 
    assertEquals("From: Pelaajat line: 43", false, pelaajat.anna(1) == p1); 
    assertEquals("From: Pelaajat line: 44", true, pelaajat.anna(1) == p2); 
    try {
    assertEquals("From: Pelaajat line: 45", p2, pelaajat.anna(3)); 
    fail("Pelaajat: 45 Did not throw IndexOutOfBoundsException");
    } catch(IndexOutOfBoundsException _e_){ _e_.getMessage(); }
    pelaajat.lisaa(p1); assertEquals("From: Pelaajat line: 46", 4, pelaajat.getLkm()); 
    pelaajat.lisaa(p1); assertEquals("From: Pelaajat line: 47", 5, pelaajat.getLkm()); 
    try {
    pelaajat.lisaa(p1); 
    fail("Pelaajat: 48 Did not throw SailoException");
    } catch(SailoException _e_){ _e_.getMessage(); }
  } // Generated by ComTest END
}